contrib/epee/include/string_tools.h

```diff
@@ -353,7 +353,12 @@ POP_WARNINGS
 	//----------------------------------------------------------------------------
 	inline bool validate_hex(uint64_t length, const std::string& str)
 	{
-		return std::regex_match(str, std::regex("'^[0-9abcdefABCDEF]+$'")) && str.size() == length;
+		if (str.size() != length)
+			return false;
+		for (char c: str)
+			if (!strchr("0123456789abcdefABCDEF", c))
```

Comment:
I think this is an improvement over regex (which is abused a bit in epee), but what about:
```c++
for (char c : str) {
    if (!((u8'0' <= c && c <= u8'9') || (u8'a' <= c && c <= u8'f') || (u8'A' <= c && c <= u8'F')) {
        return false;
    }
}
```
`strchr` is a bit more complex because its assumes that that input string can be quite large (and is not looking for ranges). The `u8` is also a pessimism to ensure that ASCII-compatible values are being used instead of implementation defined character set.

Reply:
I was thinking of using ctype.h (isxdigit) instead actually.